lib/pos_blockchain_test.go

```diff
@@ -1471,7 +1472,139 @@ func _verifyCommitRuleHelper(testMeta *TestMeta, committedBlocks []*BlockHash, u
 	}
 }
 
-func _generateBlockAndAddToBestChain(testMeta *TestMeta, blockHeight uint64, view uint64, seed int64) *MsgDeSoBlock {
+func TestProcessBlockPoS(t *testing.T) {
+	testMeta := NewTestPoSBlockchainWithValidators(t)
+
+	// Create a bad block and try to process it.
+	dummyBlock := _generateDummyBlock(testMeta, 12, 12, 887)
+	success, isOrphan, missingBlockHashes, err := testMeta.chain.processBlockPoS(dummyBlock, 12, true)
+	require.False(t, success)
+	require.False(t, isOrphan)
+	require.Len(t, missingBlockHashes, 0)
+	require.Error(t, err)
+
+	var realBlock *MsgDeSoBlock
+	realBlock = _generateRealBlock(testMeta, 12, 12, 889, testMeta.chain.GetBestChainTip().Hash)
+	success, isOrphan, missingBlockHashes, err = testMeta.chain.processBlockPoS(realBlock, 12, true)
+	require.True(t, success)
+	require.False(t, isOrphan)
+	require.Len(t, missingBlockHashes, 0)
+	require.NoError(t, err)
+
+	// Okay now we can check the best chain.
+	// We expect the block to be uncommitted.
+	blockHash1, err := realBlock.Hash()
+	require.NoError(t, err)
+	_verifyCommitRuleHelper(testMeta, []*BlockHash{}, []*BlockHash{blockHash1}, nil)
```

Comment:
nit: similar comment as above
```suggestion
	// Test a good block with a vote QC
	{
		realBlock = _generateRealBlock(testMeta, 12, 12, 889, testMeta.chain.GetBestChainTip().Hash)
		success, isOrphan, missingBlockHashes, err = testMeta.chain.processBlockPoS(realBlock, 12, true)
		require.True(t, success)
		require.False(t, isOrphan)
		require.Len(t, missingBlockHashes, 0)
		require.NoError(t, err)

		// Okay now we can check the best chain.
		// We expect the block to be uncommitted.
		blockHash1, err := realBlock.Hash()
		require.NoError(t, err)
		_verifyCommitRuleHelper(testMeta, []*BlockHash{}, []*BlockHash{blockHash1}, nil)
	}
```